crypto/src/utils.rs

```diff
@@ -0,0 +1,212 @@
+// Copyright 2018 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// spell-checker:ignore cipherparams ciphertext
+
+#![allow(dead_code)] // TODO Remove after complete ECR-2518 and  ECR-2519
+use super::{gen_keypair, gen_keypair_from_seed, PublicKey, SecretKey, Seed, SEED_LENGTH};
+use hex_buffer_serde::Hex;
+use pwbox::{sodium::Sodium, ErasedPwBox, Eraser, Suite};
+use rand::thread_rng;
+use std::borrow::Cow;
+#[cfg(unix)]
+use std::os::unix::fs::{MetadataExt, OpenOptionsExt};
+use std::{
+    fs::{File, OpenOptions},
+    io::{Error, ErrorKind, Read, Write},
+    path::Path,
+};
+use toml;
+
+/// Creates a TOML file that contains encrypted `SecretKey` and returns `PublicKey` for the secret key.
+pub fn create_keys_file<P: AsRef<Path>>(path: P, pass_phrase: &[u8]) -> Result<PublicKey, Error> {
```

Comment:
As a slight usability improvement, maybe we could accept passphrase as `W: AsRef<[u8]>` so that you can pass there whatever you want: a slice or a `String`. Ditto for read_keys_from_file().

Reply:
Maybe `generate_keys_file` is more consistent with the other our functions like `gen_keypair`?